frontend/packages/dev-console/src/components/monitoring/metrics/MetricsChart.tsx

```diff
@@ -0,0 +1,64 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { RootState } from '@console/internal/redux';
+import { queryBrowserDeleteAllQueries } from '@console/internal/actions/ui';
+import { getActiveNamespace } from '@console/internal/reducers/ui';
+import { QueryBrowser, QueryObj } from '@console/internal/components/monitoring/query-browser';
+import { QueryTable } from '@console/internal/components/monitoring/metrics';
+import EmptyStateQuery from './EmptyStateQuery';
+import './MetricsChart.scss';
+
+type StateProps = {
+  queries: QueryObj[];
+  namespace: string;
+};
+type MetricsChartProps = {
+  deleteAll?: () => never;
+  queries: QueryObj[];
+  namespace: string;
+};
+const DEFAULT_TIME_SPAN = 30 * 60 * 1000;
+
+export const MetricsChart: React.FC<MetricsChartProps> = ({ deleteAll, queries, namespace }) => {
+  React.useEffect(() => deleteAll, [deleteAll]);
+  // Use React.useMemo() to prevent these two arrays being recreated on every render, which would trigger unnecessary
+  // re-renders of QueryBrowser, which can be quite slow
+  const queriesMemoKey = JSON.stringify(_.map(queries, 'query'));
+  const queryStrings = React.useMemo(
+    () => _.filter(_.map(queries, 'query')),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [queriesMemoKey],
+  );
+  const disabledSeriesMemoKey = JSON.stringify(
+    _.reject(_.map(queries, 'disabledSeries'), _.isEmpty),
+  );
+  const disabledSeries = React.useMemo(
+    () => _.map(queries, 'disabledSeries'),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [disabledSeriesMemoKey],
+  );
```

Comment:
Instead of essentially using a hash of the queryStrings and disabledSeries as memo keys and disabling an eslint rule, we could just use `React.useState` and `React.useEffect` to conditionally update these two values. Something like this:

```tsx
  const [queryStrings, setQueryStrings] = React.useState<QueryBrowserProps['queries']>([]);
  const [disabledSeries, setDisabledQueries] = React.useState<QueryBrowserProps['disabledSeries']>([]);
  React.useEffect(() => {
    setQueryStrings((prev) => {
      const next = _.compact(_.map(queries, 'query'));
      return _.isEqual(prev, next) ? prev : next;
    });
    setDisabledQueries((prev) => {
      const next = _.compact(_.map(queries, 'disabledSeries'));
      return _.isEqual(prev, next) ? prev : next;
    });
  }, [queries]);
```

Reply:
Thanks, have updated